daemon/global_statistics.c

```diff
@@ -2892,3 +3784,45 @@ void *global_statistics_workers_main(void *ptr)
     return NULL;
 }
 
+// ---------------------------------------------------------------------------------------------------------------------
+// sqlite3 thread
+
+static void global_statistics_sqlite3_cleanup(void *ptr)
+{
+    worker_unregister();
+
+    struct netdata_static_thread *static_thread = (struct netdata_static_thread *)ptr;
+    static_thread->enabled = NETDATA_MAIN_THREAD_EXITING;
+
+    info("cleaning up...");
+
+    static_thread->enabled = NETDATA_MAIN_THREAD_EXITED;
+}
+
+void *global_statistics_sqlite3_main(void *ptr)
+{
+    global_statistics_register_workers();
+
+    netdata_thread_cleanup_push(global_statistics_sqlite3_cleanup, ptr);
+
+            int update_every =
+                    (int)config_get_number(CONFIG_SECTION_GLOBAL_STATISTICS, "update every", localhost->rrd_update_every);
+            if (update_every < localhost->rrd_update_every)
+                update_every = localhost->rrd_update_every;
+
+            usec_t step = update_every * USEC_PER_SEC;
+            heartbeat_t hb;
+            heartbeat_init(&hb);
+
+            while (service_running(SERVICE_COLLECTORS)) {
+                worker_is_idle();
```

Comment:
```sh
daemon/global_statistics.c: 3819 in global_statistics_sqlite3_main()
3813                 usec_t step = update_every * USEC_PER_SEC;
3814                 heartbeat_t hb;
3815                 heartbeat_init(&hb);
3816     
3817                 while (service_running(SERVICE_COLLECTORS)) {
3818                     worker_is_idle();
>>>     CID 433482:  Integer handling issues  (DIVIDE_BY_ZERO)
>>>     In function call "heartbeat_next", modulo by expression "step" which may be zero has undefined behavior.
3819                     heartbeat_next(&hb, step);
3820     
3821                     worker_is_busy(WORKER_JOB_SQLITE3);
3822                     sqlite3_statistics_charts();
3823                 }
3824     
3825         netdata_thread_cleanup_pop(1);
3826         return NULL;
`` `

Reply:
Step cannot be zero. If it is, netdata should crash so that the developer can fix it.
Since coverity believes it is zero, then there is probably a calculation somewhere that may result to a zero step.
Can you find it @thiagoftsm ? Does coverity provide any info about the exact path that resulted to a zero step?

Reply:
Unfortunately this was the unique message it gave us. I will run again after last commits to verify.
We have the same feeling, but I copied here, because I though you could find a place to  `update_every` to be zero, so I will mark this as resolved.